remote scan exception halts tserver

We we seeing cases where a scan on a remote tserver (possibly for data from accumulo.root) fails and this results in the local tserver be halted.  This happened on 3 tservers at the same time running a scan on the same remote tserver.  The exception looks something like this:
```
org.apache.accumulo.core.util.threads.ThreadPools$ExecutionError: Critical scheduled background task failed.
      at org.apache.accumulo.core.util.threads.ThreadPools.checkTaskfailed(ThreadPools.java.139)
...
Caused by: java.util.concurrent.ExecutionException: java.lang.RuntimeException: org.apache.accumulo.core.clientImpl.AccumuloServerException: Error on server RemoteHost:9997
...
Caused by: java.lang.RuntimeException: org.apache.accumulo.core.clientImpl.AccumuloServerException: Error on server RemoteHost:9997
     at org.apache.accumujlo.core.clientImpl.TabletServerBatchReaderIterator.hasNext(TabletServerBatchReaderIterator.java:194)
...
Caused by: org.apache.accumulo.core.clientImpl.AccumuloServerException: Error on server RemoteHost:9997
    at org.apache.accumulo.core.clientImpl.TabletServerBatchReaderiterator.doLookup(TabletServerBatchReaderIterator.java:911)
...
```
I do not see any exceptions in the RemoteHost logs that correlate with these failures.  I see the accumulo.audit INFO message "operation: Permitted; user: <user>; client: <ip>:<port>; action: authenticate"  but I do not see any exceptions or other messages.

openjdk versio0n "11.0.15" 2022-04-19 LTS (Corretto)
CentOS 7.3.1611
Accumulo version 2.1.1-SNAPSHOT

--- a/server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java
+++ b/server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java
@@ -24,7 +24,6 @@ import static org.apache.accumulo.core.metadata.schema.TabletMetadata.ColumnType
 import static org.apache.accumulo.core.metadata.schema.TabletMetadata.ColumnType.LOGS;
 import static org.apache.accumulo.core.metadata.schema.TabletMetadata.ColumnType.PREV_ROW;
 import static org.apache.accumulo.core.util.UtilWaitThread.sleepUninterruptibly;
-import static org.apache.accumulo.core.util.threads.ThreadPools.watchCriticalFixedDelay;
 import static org.apache.accumulo.core.util.threads.ThreadPools.watchCriticalScheduledTask;
 import static org.apache.accumulo.core.util.threads.ThreadPools.watchNonCriticalScheduledTask;
 
@@ -823,7 +822,7 @@ public class TabletServer extends AbstractServer implements TabletHostingServer 
 
     long tabletCheckFrequency = aconf.getTimeInMillis(Property.TSERV_HEALTH_CHECK_FREQ);
     // Periodically check that metadata of tablets matches what is held in memory
-    watchCriticalFixedDelay(aconf, tabletCheckFrequency, () -> {
+    watchNonCriticalScheduledTask(context.getScheduledExecutor().scheduleWithFixedDelay(() -> {
       final SortedMap<KeyExtent,Tablet> onlineTabletsSnapshot = onlineTablets.snapshot();
 
       Map<KeyExtent,MetadataUpdateCount> updateCounts = new HashMap<>();
@@ -859,7 +858,7 @@ public class TabletServer extends AbstractServer implements TabletHostingServer 
       } finally {
         mdScanSpan.end();
       }
-    });
+    }, tabletCheckFrequency, tabletCheckFrequency, TimeUnit.SECONDS));
 
     final long CLEANUP_BULK_LOADED_CACHE_MILLIS = TimeUnit.MINUTES.toMillis(15);
     watchCriticalScheduledTask(context.getScheduledExecutor().scheduleWithFixedDelay(

--- a/server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java
+++ b/server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java
@@ -24,6 +24,7 @@ import static org.apache.accumulo.core.metadata.schema.TabletMetadata.ColumnType
 import static org.apache.accumulo.core.metadata.schema.TabletMetadata.ColumnType.LOGS;
 import static org.apache.accumulo.core.metadata.schema.TabletMetadata.ColumnType.PREV_ROW;
 import static org.apache.accumulo.core.util.UtilWaitThread.sleepUninterruptibly;
+import static org.apache.accumulo.core.util.threads.ThreadPools.watchCriticalFixedDelay;
 import static org.apache.accumulo.core.util.threads.ThreadPools.watchCriticalScheduledTask;
 import static org.apache.accumulo.core.util.threads.ThreadPools.watchNonCriticalScheduledTask;
 
@@ -822,7 +823,7 @@ public class TabletServer extends AbstractServer implements TabletHostingServer 
 
     long tabletCheckFrequency = aconf.getTimeInMillis(Property.TSERV_HEALTH_CHECK_FREQ);
     // Periodically check that metadata of tablets matches what is held in memory
-    watchNonCriticalScheduledTask(context.getScheduledExecutor().scheduleWithFixedDelay(() -> {
+    watchCriticalFixedDelay(aconf, tabletCheckFrequency, () -> {
       final SortedMap<KeyExtent,Tablet> onlineTabletsSnapshot = onlineTablets.snapshot();
 
       Map<KeyExtent,MetadataUpdateCount> updateCounts = new HashMap<>();
@@ -858,7 +859,7 @@ public class TabletServer extends AbstractServer implements TabletHostingServer 
       } finally {
         mdScanSpan.end();
       }
-    }, tabletCheckFrequency, tabletCheckFrequency, TimeUnit.SECONDS));
+    });
 
     final long CLEANUP_BULK_LOADED_CACHE_MILLIS = TimeUnit.MINUTES.toMillis(15);
     watchCriticalScheduledTask(context.getScheduledExecutor().scheduleWithFixedDelay(

--- a/server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java
+++ b/server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java
@@ -841,7 +841,6 @@ public class TabletServer extends AbstractServer implements TabletHostingServer 
         try (TabletsMetadata tabletsMetadata =
             getContext().getAmple().readTablets().forTablets(onlineTabletsSnapshot.keySet())
                 .fetch(FILES, LOGS, ECOMP, PREV_ROW).build()) {
-          mdScanSpan.end();
           duration = Duration.between(start, Instant.now());
           log.debug("Metadata scan took {}ms for {} tablets read.", duration.toMillis(),
               onlineTabletsSnapshot.keySet().size());
@@ -854,6 +853,11 @@ public class TabletServer extends AbstractServer implements TabletHostingServer 
             tablet.compareTabletInfo(counter, tabletMetadata);
           }
         }
+      } catch (Exception e) {
+        log.error("Unable to complete verification of tablet metadata", e);
+        TraceUtil.setException(mdScanSpan, e, true);
+      } finally {
+        mdScanSpan.end();
       }
     });
 

